llama-index-integrations/readers/llama-index-readers-file/llama_index/readers/file/unstructured/base.py

```diff
@@ -98,6 +101,14 @@ def load_data(
         """
         unstructured_kwargs = unstructured_kwargs.copy() if unstructured_kwargs else {}
 
+        if (
+            unstructured_kwargs.get("file") is not None
+            and unstructured_kwargs.get("metadata_filename") is None
+        ):
+            raise ValueError(
+                "Please provide a 'metadata_filename' as part of the 'unstructured_kwargs' when loading a file stream."
+            )
+
```

Comment:
When loading a file stream, we don't have a filename in the metadata, so we require the user to provide one so that we always have valid doc_ids!